aiida_common_workflows/workflows/bands/fleur/generator.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""Implementation of `aiida_common_workflows.common.bands.generator.CommonBandsInputGenerator` for Fleur."""
+
+from aiida import engine, orm
+
+from aiida_common_workflows.generators import CodeType
+
+from ..generator import CommonBandsInputGenerator
+
+__all__ = ('FleurCommonBandsInputGenerator',)
+
+
+class FleurCommonBandsInputGenerator(CommonBandsInputGenerator):
+    """Generator of inputs for the FleurCommonBandsWorkChain"""
+
+    @classmethod
+    def define(cls, spec):
+        """Define the specification of the input generator.
+
+        The ports defined on the specification are the inputs that will be accepted by the ``get_builder`` method.
+        """
+        super().define(spec)
+        spec.inputs['engines']['bands']['code'].valid_type = CodeType('fleur.fleur')
+
+    def _construct_builder(self, **kwargs) -> engine.ProcessBuilder:
+        """Construct a process builder based on the provided keyword arguments.
+
+        The keyword arguments will have been validated against the input generator specification.
+        """
+        # pylint: disable=too-many-branches,too-many-statements,too-many-locals
+        engines = kwargs.get('engines', None)
+        parent_folder = kwargs['parent_folder']
+        bands_kpoints = kwargs['bands_kpoints']
+
+        # From the parent folder, we retrieve the calculation that created it.
+        parent_calc = parent_folder.creator
+        if parent_calc.process_type != 'aiida.calculations:fleur.fleur':
+            raise ValueError('The `parent_folder` has not been created by a FleurCalculation')
+        builder_parent = parent_folder.creator.get_builder_restart()
+
+        #Transfer inputs from the parent calculation
+        builder = self.process_class.get_builder()
+        builder.options = orm.Dict(dict=builder_parent.metadata.options)
+        builder.fleur = builder_parent.code
```

Comment:
Not really a point for this PR, but I don't think it is advisable to have the `Code` and `options` inputs a separate ports. And especially exposing the `options` as a `Dict` node instead of a normal `dict` in the metadata of the calculation job is subideal. Is there a reason why you do this and you don't let the caller specify this information in the exposed inputs of the `FleurScfWorkChain`?

Taking a closer look, it seems that this is just a continuation problem of the fact that `FleurScfWorkChain` doesn't expose the inputs of `FleurBaseWorkChain` but manually creates input ports. If possible, I would recommend to properly use `expose_inputs` when wrapping subworkflows.

Reply:
Yes you are right we have definitely underused exposing the inputs/outputs in the `aiida-fleur` plugin. I've been trying to expand the usage of this, where it leads to no breaking changes in the last months. One slight problem is that on the level of the `FleurSCFWorkChain` we have two different codes that can potentially be used. The `fleur` code and the `inpgen` code, which is used when we start a calculation from a structure input.

We can probably move to exposing the inputs by exposing both the `FleurBaseWorkChain` and the `FleurinputgenCalculation`, with one or both being in a namespace. But since the scf workchain is the main workchain that is then used throughout almost all higher level workchains this is quite a large breaking change, so we need to be careful with it.